py/visdom/__init__.py

```diff
@@ -281,6 +282,19 @@ def __init__(
 
         if send:  # if you're talking to a server, get a backchannel
             self.setup_socket()
+        # Wait for initialization before starting
+        time_spent = 0
+        while self.use_socket and not self.socket_alive and time_spent < 5:
+            time.sleep(0.1)
```

Comment:
It is not really important, but this is typically implemented by doubling the sleep time each time you sleep.

Reply:
Sounds good to me, have done that in the past but didn't think about it here.